Mlem/App/Globals/Definitions/Palette.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ColorProvider.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-05-06.
+//
+
+import Foundation
+import SwiftUI
+
+protocol PaletteProviding {
+    // basics
+    var primary: Color { get }
+    var background: Color { get }
+    var secondaryBackground: Color { get }
+    var tertiaryBackground: Color { get }
+    var accent: Color { get }
+    var uiAccent: UIColor { get }
+    
+    // interactions
+    var upvote: Color { get }
+    var downvote: Color { get }
+    var save: Color { get }
+}
+
+enum PaletteOption: String {
+    case standard, monochrome
+    
+    var palette: any PaletteProviding {
+        switch self {
+        case .standard:
+            StandardPalette()
+        case .monochrome:
+            MonochromePalette()
+        }
+    }
+}
```

Comment:
Instead of defining separate `StandardPalette` and `MonochromePalette` structs, maybe it would be better to define a single `ColorPalette` struct like so, with instances of the struct defined statically to represent each palette?

```swift
struct ColorPalette: PaletteProviding {
    let accent: Color
    // etc
    
    static let standard: Self = .init(accent: .blue)
    static let monochrome: Self = .init(accent: .primary)
}
```

Reply:
Yeah, that seems better--it looks like `static` is implicitly `lazy`, too, so it should be equally performant